### ISSUE TITLE
Selector seems to cause subscribers to go out of sync.

### ISSUE THREAD

**Issue by andersonmcook** [commanded/eventstore#128]
I think `selector` may be broken.

I've created a trivial example at the bottom to hopefully demonstrate what's going on.

What I'm trying to accomplish is to perform a query or write inside of `handle_info({:events, events}, state)` but before `ack`ing.  The thought being that if it could not accomplish the work on the event, it should not advance to the next event. I'm processing the events as they come in.

I've created a `selector` for events that I never wish to process.  Anything that passes through it, will cause some work to be done.

However, if `handle_info` processes too long before the `ack`, then it never receives more events.
Running a query that takes some amount of time inside of `handle_info` gets the same behavior.
I've found that it can be as low as 10 milliseconds if using `Process.sleep`.

I get the same behavior if I put the work inside of a `Task`.

Sending more events to the subscriber does nothing until it reaches max capacity, in which case, I receive a warning per event, but still no further processing.

In the example, all of the above is using a `selector` that returns `true` for odd numbers and `false` for even numbers.  So I'll automatically `ack` even numbers, but do some work that takes time on odd numbers before `ack`ing.

When I remove the `selector`, everything works fine.

```elixir
defmodule TestEvent do
  defstruct [:n]
end
```

For a `Range` of 1 to 100, I'll append each individual event to the stream giving it that number as a payload.

If I append them in a batch, there's no problem processing them all.  It's only when they are appended separately that the subscriber almost immediately goes out of sync and won't recover until it is restarted.

```elixir
defmodule TestSubscriber do
  import Integer, only: [is_odd: 1]

  def start_link do
    state = %{
      stream_id: "TestStream",
      subscription: nil,
      subscription_name: Atom.to_string(__MODULE__)
    }

    GenServer.start_link(__MODULE__, state, name: __MODULE__)
  end

  def init(state) do
    {:ok, subscription} =
      EventStore.subscribe_to_stream(
        state.stream_id,
        state.subscription_name,
        self(),
        selector: &test_selector/1
      )

    {:ok, %{state | subscription: subscription}}
  end

  def handle_info({:events, events}, state) do
    # simulate work
    Process.sleep(10)
    :ok = EventStore.ack(state.subscription, events)

    {:noreply, state}
  end

  def handle_info(_request, state), do: {:noreply, state}

  def test_selector(%{data: %{n: n}}) when is_odd(n) do
    true
  end

  def test_selector(_) do
    false
  end
end
```

**Comment by slashdotdash** [comment]
Thanks for the bug report @andersonmcook I wil take a look.

**Comment by slashdotdash** [comment]
@andersonmcook I've merged a fix for this issue, if you get the chance could you try the `master` branch and confirm it works.

**Comment by slashdotdash** [comment]
@andersonmcook I've just published v0.15.1 to Hex containing the fix for this issue.
